src/pages/iou/ReceiptSelector/NavigationAwareCamera.js

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useState} from 'react';
+import {Camera} from 'react-native-vision-camera';
+import {useTabAnimation} from '@react-navigation/material-top-tabs';
+import {useNavigation} from '@react-navigation/native';
+import refPropTypes from '../../../components/refPropTypes';
+
+const propTypes = {
+    /* Forwarded ref */
+    forwardedRef: refPropTypes.isRequired,
+};
+
+// Wraps a camera that will only be active when the tab is focused or as asoon as it start to become focused.
+function NavigationAwareCamera(props) {
```

Comment:
No, this would defy the performance optimization of activating the camera as soon as possible. This would only start activating the camera once the page is fully visible. Now the camera activates while we are still animating to the camera page.

Reply:
And in addition to that this would cause a re-render of the whole ReceiptSelector once we change a tab, which isn't great for performance either. We really need to incorporate in our work on the react code base, that state that only belongs to a sub component gets bundled with that sub component into a separate component to avoid performance losses (e.g. re-rendering a whole screen only because one child part needs to update)